pkg/controllers/customcluster_controller.go

```diff
@@ -47,16 +77,392 @@ type CustomClusterController struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
 func (r *CustomClusterController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	log := ctrl.LoggerFrom(ctx)
+
+	// Fetch the customCluster instance.
+	customCluster := &v1alpha1.CustomCluster{}
+	if err := r.Client.Get(ctx, req.NamespacedName, customCluster); err != nil {
+		if apierrors.IsNotFound(err) {
+			return ctrl.Result{Requeue: false}, nil
+		}
+		return ctrl.Result{RequeueAfter: RequeueAfter}, err
```

Comment:
This is waiting for CustomCluster to occur by retrying. But we need event trigger here